src/flightcheck/pipeline.js

```diff
@@ -0,0 +1,178 @@
+    assert(this.build['tag'], 'Pipeline needs a tag to checkout')
+
+    // Try to determine optional variables
+    if (this.build.source == null) {
+      if (this.build.repo.indexOf('github.com') !== -1) {
+        this.build.source = 'github'
+      }
+    }
+
+    if (this.build.source === 'github') {
+      // Filter the github information from the repo url
+      // Possible urls are https://github.com/vocalapp/vocal
+      // and git@github.com:vocalapp/vocal.git
+      const splits = this.build.repo.split(/(\/|:)/)
+      const owner = splits[splits.length - 3].replace('.', '_')
+      const repo = splits[splits.length - 1].replace('.git', '').replace('.', '_')
+
+      this.build.name = `com.github.${owner}.${repo}`
+    }
+
+    if (this.build.version == null) {
+      this.build.version = semver.valid(this.build.tag)
+    }
+
+    // Check optional variables to make sure we can determine them from given manditory variables
+    if (build.version == null && this.build.version == null) {
+      throw new Error('Pipeline cannot parse tag into valid semver. Please specify it manually')
+    } else if (semver.valid(this.build.version) == null) {
+      throw new Error('Pipeline cannot parse the given version to valid semver')
+    }
+
+    assert.equal(this.build['source'], 'github', 'Pipeline needs a source to build from')
+    assert(this.build['name'], 'Pipeline was unable to parse repo for valid name. Please specify it manually')
+
+    // Setup some dynamic variables
+    this.build.dir = path.join(config.flightcheck.directory, 'projects', this.build.name)
+  }
+
+  /**
+   * setup
+   * Clones repo and sets up workspace for pipeline
+   *
+   * @return {Void}
+   */
+  async setup () {
+    const repoFolder = path.join(this.build.dir, 'repository')
+
+    // TODO: if there is a repo folder, check to make sure its correct branch or delete it and start again
+    await fsHelper.mkdirp(repoFolder)
+    const repo = await git.Clone(this.build.repo, repoFolder)
+    await repo.checkoutBranch(this.build.tag)
+  }
+
+  /**
+   * teardown
+   * Deletes all remaining files in workspace and other cleanup tasks
+   *
+   * @return {Void}
+   */
+  async teardown () {
+    if (config.env !== 'development') {
+      await fsHelper.rmp(this.build.dir)
+      return
+    }
+
+    // TODO: This could probably be removed once the repo checking (@see setup) is finished
+    log.warn('Flightcheck is running in development mode. Only deleting workspace repo.')
+    await fsHelper.rmp(path.join(this.build.dir, 'repository'))
+  }
+
+  /**
+   * run
+   * Runs the pipeline process
+   *
+   * @returns {Object} - {
+   *   {Number} errors - number of errors the hooks aquired
+   *   {Number} warnings - number of warnings the hook aquired
+   *   {Object} information - information to be updated in the database
+   *   {Array} issues - generated issues for GitHub with title and body
+   * }
+   */
+  async run () {
```

Comment:
`async run` cannot be changed until the pipes are changed. This code is a direct copy from how it's ran now.